rest/method_item_put.go

```diff
@@ -51,13 +51,20 @@ func itemPut(ctx context.Context, r *http.Request, route *RouteMatch) (status in
 	status = 200
 	var changes map[string]interface{}
 	var base map[string]interface{}
+	var errs map[string][]interface{}
 	if original == nil {
 		// PUT used to create a new document.
-		changes, base = rsrc.Validator().Prepare(ctx, payload, nil, false)
+		changes, base, errs = rsrc.Validator().Prepare(ctx, payload, nil, false)
+		if len(errs) > 0 {
+			return 422, nil, &Error{422, "Document contains error(s)", errs}
+		}
 		status = 201
 	} else {
 		// PUT used to replace an existing document.
-		changes, base = rsrc.Validator().Prepare(ctx, payload, &original.Payload, true)
+		changes, base, errs = rsrc.Validator().Prepare(ctx, payload, &original.Payload, true)
+		if len(errs) > 0 {
+			return 422, nil, &Error{422, "Document contains error(s)", errs}
```

Comment:
One feedbak I got... Since we do not validate _all_ fields as parts of prepare (e.g. only fields present), I suppose this means we are returning a potentially _partial_ error response, right?

Reply:
Well, yes. The remaining validation will be done by `schema.Validate()`. I guess we can fix this in the future rewrite of this part of the system.

Reply:
Sorry for the delay. I think I may have a proposal that is better.

Maybe we could handle the errors in prepare instead of returning it, by assuming that if an input value fails validation, we treat it as _changed_, and places the original input value in `changes` without a DeepEqual test. We document that in a comment. Then we get to return all the errors to the user, because the same value would fail `schema.Validate()` later anyway.